kong/tools/responses.lua

```diff
@@ -61,6 +61,8 @@ local function send_response(status_code)
         ngx.log(ngx.ERR, tostring(content))
       end
       ngx.ctx.stop_phases = true -- interrupt other phases of this request
+    elseif status_code == _M.status_codes.HTTP_UNAUTHORIZED then
+      ngx.header["WWW-Authenticate"] = "Key realm=\""..constants.NAME.."\"" -- respond with a challenge
```

Comment:
@thibaultCha definitely, I'm so new to lua/kong land. Just made this PR after 30 mins of going through code, I'll update the PR soon, thanks so much for pointing in the right direction!